### PR TITLE
Fix example 24 with enveloped credential

### LINKED ISSUE #24: Speculations about future JWT extensions
A note in the spec says:
> Implementers are warned that JWTs are not capable of encoding multiple subjects and are thus not capable of encoding a verifiable credential with more than one subject. JWTs might support multiple subjects in the future and implementers are advised to refer to the [JSON Web Token Claim Registry](https://www.iana.org/assignments/jwt/) for multi-subject JWT claim names or the [Nested JSON Web Token](https://datatracker.ietf.org/doc/draft-yusef-oauth-nested-jwt/) specification.

Speculating about how other specs might or might be extended seems unhelpful and inappropriate.  Also, the "not capable" statement is overly strong, as JWTs could be extended to do so.  I suggest replacing the note above with:
> At the time of this writing, JWTs do not have a representation for multiple subjects and are thus not capable of encoding a verifiable credential with more than one subject. Should the working group decide to do so, we could create a new JWT claim for this purpose and register it in the [JSON Web Token Claims Registry](https://www.iana.org/assignments/jwt/).

Attn: @OR13

### LINKED ISSUE #217: Example 24 is invalid
[Example 24](https://www.w3.org/TR/vc-jose-cose/#example-presentation) uses the `verifiableCredential` property to refer to several credentials. However, the [formal definition of this property](https://www.w3.org/TR/vc-jose-cose/#example-presentation) says:

> The value MUST be one or more [verifiable credential](https://www.w3.org/TR/vc-data-model-2.0/#dfn-verifiable-credential) and/or [enveloped verifiable credential](https://www.w3.org/TR/vc-data-model-2.0/#enveloped-verifiable-credentials) objects (to be clear, the values MUST NOT be non-object values such as numbers, strings, ***or URLs***).

(Emphasis is mine.). In other words, these lines of the example:

```json
    "https://vendor.example/credentials/42", 
    "did:example:123",
    "urn:uuid:01ec9426-c175-4e39-a006-d30050e28214",
    "urn:ietf:params:oauth:jwk-thumbprint:sha-256:_Fpfe27AuGmEljZE9s2lw2UH-qrZLRFNrWbJrWIe4SI",
    "data:application/vc+ld+json+sd-jwt;…"
```

are all forbidden per the VCDM spec and must be removed. (Or exchanged for a bona fide JSON object for another credential.)

### PR DESCRIPTION
**Pull request by decentralgabe**
Fix #217

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-jose-cose/pull/219.html" title="Last updated on Jan 29, 2024, 9:07 PM UTC (813298b)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-jose-cose/219/3c476fc...813298b.html" title="Last updated on Jan 29, 2024, 9:07 PM UTC (813298b)">Diff</a>

### CODE REVIEW

**Review by iherman — CHANGES_REQUESTED** [review]
Unfortunately, Github does not let me to comment directly on lines that were not edited. I believe the example should be:

```json
{ 
  "@context": ["https://www.w3.org/ns/credentials/v2"], 
  "type": ["VerifiablePresentation"], 
  "holder": "urn:ietf:params:oauth:jwk-thumbprint:sha-256:_Fpfe27AuGmEljZE9s2lw2UH-qrZLRFNrWbJrWIe4SI", 
  "verifiableCredential": [{ 
    "@context": ["https://www.w3.org/ns/credentials/v2"], 
    "type": ["EnvelopedVerifiableCredential"], 
    "id": "data:application/vcld+json+sd-jwt;eyJhbGciOiJFUzM4NCIsImtpZCI6IlNJM1JITm91aDhvODFOT09OUFFVQUw3RWdaLWtJNl94ajlvUkV2WDF4T3ciLCJ0eXAiOiJ2YytsZCtqc29uK3NkLWp3dCIsImN0eSI6InZjK2xkK2pzb24ifQ.eyJAY29udGV4dCI6WyJodHRwczovL3d3dy53My5vcmcvbnMvY3JlZGVudGlhbHMvdjIiLCJodHRwczovL3d3dy53My5vcmcvbnMvY3JlZGVudGlhbHMvZXhhbXBsZXMvdjIiXSwiaXNzdWVyIjoiaHR0cHM6Ly91bml2ZXJzaXR5LmV4YW1wbGUvaXNzdWVycy81NjUwNDkiLCJ2YWxpZEZyb20iOiIyMDEwLTAxLTAxVDE5OjIzOjI0WiIsImNyZWRlbnRpYWxTY2hlbWEiOnsiX3NkIjpbIkU3dU1sSWFyS29iYXJTdEZGRjctZm5qaV9sQVdnM3BGMkV5dVc4dWFYakUiLCJYelRaSVgyNGdDSWxSQVFHclFoNU5FRm1XWkQtZ3Z3dkIybzB5Y0FwNFZzIl19LCJjcmVkZW50aWFsU3ViamVjdCI6eyJkZWdyZWUiOnsibmFtZSI6IkJhY2hlbG9yIG9mIFNjaWVuY2UgYW5kIEFydHMiLCJfc2QiOlsiT3oxUEZIMG0tWk9TdEhwUVZyeGlmVlpKRzhvNmlQQmNnLVZ2SXQwd2plcyJdfSwiX3NkIjpbIkVZQ1daMTZZMHB5X1VNNzRHU3NVYU9zT19mdDExTlVSaFFUTS1TT1lFTVEiXX0sIl9zZCI6WyJqT055NnZUbGNvVlAzM25oSTdERGN3ekVka3d2R3VVRXlLUjdrWEVLd3VVIiwid21BdHpwc0dRbDJveS1PY2JrSEVZcE8xb3BoX3VYcWVWVTRKekF0aFFibyJdLCJfc2RfYWxnIjoic2hhLTI1NiIsImlzcyI6Imh0dHBzOi8vdW5pdmVyc2l0eS5leGFtcGxlL2lzc3VlcnMvNTY1MDQ5IiwiaWF0IjoxNjk3Mjg5OTk2LCJleHAiOjE3Mjg5MTIzOTYsImNuZiI6eyJqd2siOnsia3R5IjoiRUMiLCJjcnYiOiJQLTM4NCIsImFsZyI6IkVTMzg0IiwieCI6InZFdV84WGxZT0ZFU2hTcVRpZ2JSYWduZ0ZGM1p5U0xrclNHekh3azFBT1loanhlazVhV21HY2UwZU05S0pWOEIiLCJ5IjoiRUpNY2czWXBzUTB3M2RLNHlVa25QczE1Z0lsY2Yyay03dzFKLTNlYlBiOERENmQtUkhBeGUwMDkzSWpfdTRCOSJ9fX0.rYzbxb6j1dwop8_s491iArVVJNm6A6C3b742gOm_qYO3zdkyQU4_VxxOSJ8ECcmWj2r5KyiCNC1ojfO4Yms-zBsjt7PoMYpYWBplsqXpiIvnehmM7D0eOLi40uHXki0X~WyJSWTg1YTZNMmEwX3VDWlFTVGZmTFdRIiwgImlkIiwgImh0dHA6Ly91bml2ZXJzaXR5LmV4YW1wbGUvY3JlZGVudGlhbHMvMTg3MiJd~WyJMeG5GYTBXVm8wRUluVy1QdS1fd1dRIiwgInR5cGUiLCBbIlZlcmlmaWFibGVDcmVkZW50aWFsIiwgIkV4YW1wbGVBbHVtbmlDcmVkZW50aWFsIl1d~WyJUQVdrakpCaVpxdC1rVU54X1EweUJBIiwgImlkIiwgImh0dHBzOi8vZXhhbXBsZS5vcmcvZXhhbXBsZXMvZGVncmVlLmpzb24iXQ~WyJTd2xuZFpPZzZEZ1ZERFp5X0RvYVFBIiwgInR5cGUiLCAiSnNvblNjaGVtYSJd~WyJuSnJlU3E1Nzg3RGZMSDJCbU03cXFRIiwgImlkIiwgImRpZDpleGFtcGxlOjEyMyJd~WyIxMjNNd3hNcHRiek02YUk2aW03ME1RIiwgInR5cGUiLCAiQmFjaGVsb3JEZWdyZWUiXQ" 
  }] 
}
```

All the other properties are defined on a Verifiable Credential only, and not on an Enveloped one. (I cannot check whether the value of the Data URI is correct, I let this to you...

cc: @msporny

File: index.html
Comment (iherman) [inline index.html:1632]:
```suggestion
      "id": "data:application/vc+ld+json+sd-jwt;eyJhbGciOiJFUzM4NCIsImtpZCI6IlNJM1JITm91aDhvODFOT09OUFFVQUw3RWdaLWtJNl94ajlvUkV2WDF4T3ciLCJ0eXAiOiJ2YytsZCtqc29uK3NkLWp3dCIsImN0eSI6InZjK2xkK2pzb24ifQ.eyJAY29udGV4dCI6WyJodHRwczovL3d3dy53My5vcmcvbnMvY3JlZGVudGlhbHMvdjIiLCJodHRwczovL3d3dy53My5vcmcvbnMvY3JlZGVudGlhbHMvZXhhbXBsZXMvdjIiXSwiaXNzdWVyIjoiaHR0cHM6Ly91bml2ZXJzaXR5LmV4YW1wbGUvaXNzdWVycy81NjUwNDkiLCJ2YWxpZEZyb20iOiIyMDEwLTAxLTAxVDE5OjIzOjI0WiIsImNyZWRlbnRpYWxTY2hlbWEiOnsiX3NkIjpbIkU3dU1sSWFyS29iYXJTdEZGRjctZm5qaV9sQVdnM3BGMkV5dVc4dWFYakUiLCJYelRaSVgyNGdDSWxSQVFHclFoNU5FRm1XWkQtZ3Z3dkIybzB5Y0FwNFZzIl19LCJjcmVkZW50aWFsU3ViamVjdCI6eyJkZWdyZWUiOnsibmFtZSI6IkJhY2hlbG9yIG9mIFNjaWVuY2UgYW5kIEFydHMiLCJfc2QiOlsiT3oxUEZIMG0tWk9TdEhwUVZyeGlmVlpKRzhvNmlQQmNnLVZ2SXQwd2plcyJdfSwiX3NkIjpbIkVZQ1daMTZZMHB5X1VNNzRHU3NVYU9zT19mdDExTlVSaFFUTS1TT1lFTVEiXX0sIl9zZCI6WyJqT055NnZUbGNvVlAzM25oSTdERGN3ekVka3d2R3VVRXlLUjdrWEVLd3VVIiwid21BdHpwc0dRbDJveS1PY2JrSEVZcE8xb3BoX3VYcWVWVTRKekF0aFFibyJdLCJfc2RfYWxnIjoic2hhLTI1NiIsImlzcyI6Imh0dHBzOi8vdW5pdmVyc2l0eS5leGFtcGxlL2lzc3VlcnMvNTY1MDQ5IiwiaWF0IjoxNjk3Mjg5OTk2LCJleHAiOjE3Mjg5MTIzOTYsImNuZiI6eyJqd2siOnsia3R5IjoiRUMiLCJjcnYiOiJQLTM4NCIsImFsZyI6IkVTMzg0IiwieCI6InZFdV84WGxZT0ZFU2hTcVRpZ2JSYWduZ0ZGM1p5U0xrclNHekh3azFBT1loanhlazVhV21HY2UwZU05S0pWOEIiLCJ5IjoiRUpNY2czWXBzUTB3M2RLNHlVa25QczE1Z0lsY2Yyay03dzFKLTNlYlBiOERENmQtUkhBeGUwMDkzSWpfdTRCOSJ9fX0.rYzbxb6j1dwop8_s491iArVVJNm6A6C3b742gOm_qYO3zdkyQU4_VxxOSJ8ECcmWj2r5KyiCNC1ojfO4Yms-zBsjt7PoMYpYWBplsqXpiIvnehmM7D0eOLi40uHXki0X~WyJSWTg1YTZNMmEwX3VDWlFTVGZmTFdRIiwgImlkIiwgImh0dHA6Ly91bml2ZXJzaXR5LmV4YW1wbGUvY3JlZGVudGlhbHMvMTg3MiJd~WyJMeG5GYTBXVm8wRUluVy1QdS1fd1dRIiwgInR5cGUiLCBbIlZlcmlmaWFibGVDcmVkZW50aWFsIiwgIkV4YW1wbGVBbHVtbmlDcmVkZW50aWFsIl1d~WyJUQVdrakpCaVpxdC1rVU54X1EweUJBIiwgImlkIiwgImh0dHBzOi8vZXhhbXBsZS5vcmcvZXhhbXBsZXMvZGVncmVlLmpzb24iXQ~WyJTd2xuZFpPZzZEZ1ZERFp5X0RvYVFBIiwgInR5cGUiLCAiSnNvblNjaGVtYSJd~WyJuSnJlU3E1Nzg3RGZMSDJCbU03cXFRIiwgImlkIiwgImRpZDpleGFtcGxlOjEyMyJd~WyIxMjNNd3hNcHRiek02YUk2aW03ME1RIiwgInR5cGUiLCAiQmFjaGVsb3JEZWdyZWUiXQ"
```

Comment (iherman) [reply index.html:1632]: (In case the change disappears in the long string: the Media type at the beginning should be `vc+ld+...` and not `vcld+ld+...`)

**Comment by decentralgabe** [comment]
@iherman should be good now!

**Review by OR13 — APPROVED** [review]

File: index.html
Comment (iherman) [inline index.html:1627]:
```suggestion
      
```

**Review by iherman — CHANGES_REQUESTED** [review]
You forgot to remove the `issuer` property...

I approve, provided that change is accepted.

**Comment by Sakurann** [comment]
I think it would be a good idea to fix not just the example, but enhance the description/requirements too. because the text leading to the examples was not clear either.

**Review by msporny — APPROVED** [review]

**Comment by decentralgabe** [comment]
Going to change two things:

1. Add an example of a secure enveloped credential presentation
2. Add text / descriptions

**Comment by decentralgabe** [comment]
Updated as described. The first example will render using [Orie's plugin](https://github.com/transmute-industries/respec-vc-jwt) and the second is the original example that will render as-is.

Separately I have opened [this issue](https://github.com/w3c/respec-vc/issues/9) related to #214 which will fix how these examples get rendered. Should be good to merge on approval.

**Review by brentzundel — APPROVED** [review]

**Review by iherman — COMMENTED** [review]
I do not understand. While, from my point of view, example 24 looks o.k., I do not see any difference between examples 24 and 25, except that one is coloured and the other is not. What am I missing?
Screenshot 2024-02-08 at 17 15 03

**Comment by iherman** [comment]
> @iherman this has to do with the rendering issue that we are working to fix. the first will be rendered with multiple tabs showing issued/signed/revealed/etc the second will appear as is. I'd prefer to merge this as the rendering fix is a parallel effort.
> 

Ok. But then we should probably wait with the merge of this PR to get #214 settled, so that we would get a full picture.

**Review by selfissued — APPROVED** [review]

**Comment by decentralgabe** [comment]
this will be rebased and merged after #241 which is blocked by #239

**Pull request closed without merging**
